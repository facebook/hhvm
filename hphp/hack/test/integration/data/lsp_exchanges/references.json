[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 11,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 6,
        "character": 10
      },
      "context": {
        "includeDeclaration": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 19,
        "character": 27
      },
      "context": {
        "includeDeclaration": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 13,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 37,
        "character": 3
      },
      "context": {
        "includeDeclaration": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 14,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 45,
        "character": 13
      },
      "context": {
        "includeDeclaration": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
