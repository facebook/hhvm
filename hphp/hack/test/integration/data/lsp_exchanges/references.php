<?hh //partial

class C {
  private function ref_test_1 ($x) {  // Should not match
    $y = $x;           // Should not match
    static $x = 123;   // Should match
    $z = $x;           // Request ID 11: Looking for this $x // Should match
  }

  public function ref_test_method(): int {
    return 0;
  }
}

interface IRefTestRequireExtends {
  require extends C;
}

function ref_test_require_extends(IRefTestRequireExtends $req_extends_test) {
  $req_extends_test->ref_test_method(); // Request ID 12
}

function j(int $x): int {
  return $x;
}

function h(string $x): int {
  return 5;
}

enum RefTestEnum: int {  // Request ID 14 on RefTestEnum
  SMALL = 0;
  MEDIUM = 1;
  LARGE = 2;
}

function ref_test_2(int $x): void {   //  Should match
  $x = 3; // Request ID 13: Looking for this $x   //  Should match
  j($x) + $x + h("\$x = $x");         //  First, second, and fourth should match
  $lambda1 = $x ==> $x + 1;           //  Should not match
  $lambda2 = $a ==> $x + $a;          //  Should match
  $lambda3 = function($x) {           //  Should not match
    return $x + 1; };                 //  Should not match
  $lambda4 = function($b) use($x) {   //  Should match
    return $x + $b; };                //  Should match
  $size = RefTestEnum::SMALL;
  $shape = shape(RefTestEnum::SMALL => 123);
}
