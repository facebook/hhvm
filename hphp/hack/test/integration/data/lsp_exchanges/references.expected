[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "implementationProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 11,
    "result": [
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 3,
            "character": 31
          },
          "end": {
            "line": 3,
            "character": 33
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 4,
            "character": 9
          },
          "end": {
            "line": 4,
            "character": 11
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 6,
            "character": 9
          },
          "end": {
            "line": 6,
            "character": 11
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "result": [
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 9,
            "character": 18
          },
          "end": {
            "line": 9,
            "character": 33
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 19,
            "character": 21
          },
          "end": {
            "line": 19,
            "character": 36
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 13,
    "result": [
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 36,
            "character": 24
          },
          "end": {
            "line": 36,
            "character": 26
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 37,
            "character": 2
          },
          "end": {
            "line": 37,
            "character": 4
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 38,
            "character": 4
          },
          "end": {
            "line": 38,
            "character": 6
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 38,
            "character": 10
          },
          "end": {
            "line": 38,
            "character": 12
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 38,
            "character": 24
          },
          "end": {
            "line": 38,
            "character": 26
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 40,
            "character": 20
          },
          "end": {
            "line": 40,
            "character": 22
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 43,
            "character": 30
          },
          "end": {
            "line": 43,
            "character": 32
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 44,
            "character": 11
          },
          "end": {
            "line": 44,
            "character": 13
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 14,
    "result": [
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 30,
            "character": 5
          },
          "end": {
            "line": 30,
            "character": 16
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 46,
            "character": 17
          },
          "end": {
            "line": 46,
            "character": 28
          }
        }
      },
      {
        "uri": "file://${root_path}/references.php",
        "range": {
          "start": {
            "line": 45,
            "character": 10
          },
          "end": {
            "line": 45,
            "character": 21
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]
